inc/vcf/fixer.hpp

```diff
@@ -161,6 +149,67 @@ namespace ebi
             util::writeline(output, *line);
             ignored_errors++;
         }
+
+        /**
+         * explanation of the fix:
+         * - in error.get_field there will be the field name as it appears in the FORMAT column, e.g. "GT".
+         * - if error.get_field() is empty, skip fix (copy the line as-is)
+         * - get the index in the FORMAT column of error.get_field(), e.g. in GT:AC:AN, the index of AC is 1.
+         *      if the field is not found, skip fix (copy as-is)
+         * - for each sample column
+         *      - if the field is GT
+         *          - put a "missing value" instead of the value in the index-th position after splitting by ":".
+         *          - the "missing value" is a `.` repeated `n` times, where `n` is the cardinality of the field,
+         *              with "/" as separators
+         *      - if the field is not GT
+         *          - remove it from the FORMAT and samples columns
+         * - if there were no sample columns, skip fix (copy as-is)
+         *
+         * complete example:
+         * field "GT", FORMAT: "GT:AC:AN", sample: "0/0:3,5:4"; becomes "./.:3,5:4". (index = 0, cardinality = 2)
+         * field "AC", FORMAT: "GT:AC:AN", sample: "0/0:3,5:4"; becomes "GT:AN   0/0:4". (index = 1)
+         * @param error
+         */
+        virtual void visit(SamplesFieldBodyError &error) override
+        {
+            if (error.get_field().empty()) {
```

Comment:
I saw that this check is redundant, because it is handled as if it were an invalid field, so the `if` can be safely, and I added an exception in the `SamplesFieldBodyError` constructor in case it is empty. We have the SamplesBodyError for more undefined errors in the samples columns